src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -24,6 +24,8 @@ public class DefaultModelMetadataProvider : IModelMetadataProvider
         public DefaultModelMetadataProvider(ICompositeMetadataDetailsProvider detailsProvider)
         {
             DetailsProvider = detailsProvider;
+
+            _cacheEntryFactory = CreateCacheEntry;
```

Comment:
curious: why not use `CreateCacheEntry` directly when calling `_typeCache.GetOrAdd()`?